diag_manager/diag_manager.F90

```diff
@@ -1649,13 +1649,51 @@ LOGICAL FUNCTION send_data_3d(diag_field_id, field, time, is_in, js_in, ks_in, &
     SELECT TYPE (field)
     TYPE IS (real(kind=r4_kind))
        field_out = field
+       if (use_modern_diag) then
+         allocate(real(kind=r4_kind) :: field_modern(SIZE(field,1),SIZE(field,2),SIZE(field,3),1))
```

Comment:
Can we use a pointer here instead?

Reply:
![image](https://user-images.githubusercontent.com/33458882/210074594-1b87bb67-8779-4e90-8f7b-db9bd1196a4b.png)

Reply:
@uramirez8707 the `field` needs to be contiguous.  Will it always be contiguous?

Reply:
::fingers crossed::